zcash_client_sqlite/src/lib.rs

```diff
@@ -398,18 +416,22 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
         birthday: AccountBirthday,
     ) -> Result<(AccountId, UnifiedSpendingKey), Self::Error> {
         self.transactionally(|wdb| {
-            let account = wallet::get_max_account_id(wdb.conn.0)?
+            let seed_id = HDSeedFingerprint::from_seed(seed);
+            let account_index = wallet::get_max_account_id(wdb.conn.0, &seed_id)?
                 .map(|a| a.next().ok_or(SqliteClientError::AccountIdOutOfRange))
                 .transpose()?
-                .unwrap_or(AccountId::ZERO);
+                .unwrap_or(zip32::AccountId::ZERO);
 
-            let usk = UnifiedSpendingKey::from_seed(&wdb.params, seed.expose_secret(), account)
-                .map_err(|_| SqliteClientError::KeyDerivationError(account))?;
+            let usk =
+                UnifiedSpendingKey::from_seed(&wdb.params, seed.expose_secret(), account_index)
+                    .map_err(|_| SqliteClientError::KeyDerivationError(account_index))?;
             let ufvk = usk.to_unified_full_viewing_key();
 
-            wallet::add_account(wdb.conn.0, &wdb.params, account, &ufvk, birthday)?;
+            let account_parameters = Account::Zip32(HDSeedAccount(seed_id, account_index, ufvk));
+            let account_id =
+                wallet::add_account(wdb.conn.0, &wdb.params, account_parameters, birthday)?;
```

Comment:
Aha, now I see why `Account` is documented as "inputs for creating an account" (this wasn't part of the changes extracted in #1202). This however is an internal implementation detail of `zcash_client_sqlite`, while `Account` is proposed as a general type in `zcash_client_backend`, so my comments stand.

Reply:
I hope the fixes I made to the docs and my reply to your other comment about the broader applicability of the `Account` type make sense.